contracts/src/token/erc20/extensions/wrapper.rs

```diff
@@ -0,0 +1,279 @@
+//! Extension of the ERC-20 token contract to support token wrapping.
+//!
+//! Users can deposit and withdraw "underlying tokens" and receive a matching
+//! number of "wrapped tokens". This is useful in conjunction with other
+//! modules.
+//!
+//! WARNING: Any mechanism in which the underlying token changes the {balanceOf}
+//! of an account without an explicit transfer may desynchronize this contract's
+//! supply and its underlying balance. Please exercise caution when wrapping
+//! tokens that may undercollateralize the wrapper (i.e. wrapper's total supply
+//! is higher than its underlying balance). See {_recover} for recovering value
+//! accrued to the wrapper.
+
+use alloy_primitives::{Address, U256};
+use alloy_sol_macro::sol;
+use stylus_sdk::{
+    call::Call,
+    contract, msg,
+    prelude::storage,
+    storage::{StorageAddress, TopLevelStorage},
+    stylus_proc::SolidityError,
+};
+
+use crate::token::erc20::{
+    self,
+    utils::{
+        safe_erc20::{self, ISafeErc20},
+        SafeErc20,
+    },
+    Erc20,
+};
+
+mod token {
```

Comment:
Remove this module and check how [ERC-4626](https://github.com/OpenZeppelin/rust-contracts-stylus/blob/main/contracts/src/token/erc20/extensions/erc4626.rs#L23) uses the Solidity interface.